pkg/controller/core/workload_controller.go

```diff
@@ -104,11 +138,30 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
 	case admitted:
+		var err error
+		var requeueAfter time.Duration
+		if r.podsReadyTimeout != nil {
```

Comment:
to reduce indentation, move this check to inside `admittedNotReadyWorkload` and return countingTowardsTimeout=false when not set?

Reply:
+ 1 for this, it also makes reference to *podsReadyTimeout protected by a check that `podsReadyTimeout != nil` in the same method, which makes it easier to understand. Also, added unit tests for it.